src/Model/Customer/Customers.php

```diff
@@ -0,0 +1,49 @@
+<?php
+declare(strict_types=1);
+
+namespace Billogram\Model\Customer;
+
+
+use Billogram\Exception\InvalidArgumentException;
+use Billogram\Model\CreatableFromArray;
+
+class Customers implements CreatableFromArray
+{
+    /**
+     * @var Customer[]
+     */
+    private $customers;
+
+
+    private function __construct(array $customers)
+    {
+        foreach ($customers as $customer) {
+            if (!$customers instanceof Customer) {
```

Comment:
We do not need this check. The constructor is private and we call this function from the createFromArray function